app/routes/file-upload.js

```diff
@@ -0,0 +1,54 @@
+const { checkFileExtension } = require('../utils/file-checks/extension-check')
+const { handleFileLimitExceeded } = require('../utils/file-checks/lenght-of-files-array-check')
+const { handleMessage } = require('../message')
+const { uploadFileToBlob, deleteBlob } = require('../message/upload-file-to-blob')
+
+module.exports = {
+  method: 'POST',
+  path: '/upload',
+  options: {
+    payload: {
+      output: 'stream',
+      parse: true,
+      allow: 'multipart/form-data',
+      multipart: true,
+      maxBytes: 50 * 1024 * 1024
+    }
+  },
+  handler: async (request, h) => {
+    const { payload } = request
+    const { files } = payload
+    const filesArray = Array.isArray(files) ? files : [files]
+    const results = []
+
+    try {
+      handleFileLimitExceeded(filesArray)
+
+      for (const file of filesArray) {
+        let blockBlobClient
+        try {
```

Comment:
Can you split the logic from the second try-catch into its own function which can be called on each file of the array. This keeps the route logic simple and will increase readability